### ISSUE TITLE
Check the region is `readonly` before `ReadMemory` of global variable

### ISSUE THREAD

**Issue by bin2415** [dyninst/dyninst#692]
**Describe the bug**

When I want to `simplify` an AST with `<derefOp>(<global_val>)` [here](https://github.com/dyninst/dyninst/blob/master/parseAPI/src/SymbolicExpression.C#L196), it will try to read the `global_val` content if I understand the code correctly here. 

My question is if the `global_val` is not in the readonly region such as `.data` section, the content of the `global_val` will change according to the logic of codes.

So I am wondering if it's better to check the region is `readonly` before reading content of a global variable.

**Expected behavior**
Do not read content if the global variable is in `write` region.

**Comment by mxz297** [comment]
I started with limiting to read from only readonly sections, but then I found that some compilers put jump tables in writeable sections. So, I have to allow reading writeable sections to not miss real code. 

It is definitely a security concern that compilers emit read-only data in writable sections, and I am not sure if such issues have been fixed.

**Comment by bin2415** [comment]
I met the problem in `git` program compiled by gcc O2: 

```
  46cc87:       83 3d 5a 27 23 00 05    cmpl   $0x5,0x23275a(%rip)        # 69f3e8 <push_default>
  46cc8e:       0f 87 4a 01 00 00       ja     46cdde <cmd_push+0xebe>
  46cc94:       8b 05 4e 27 23 00       mov    0x23274e(%rip),%eax        # 69f3e8 <push_default>
  46cc9a:       ff 24 c5 c0 fd 63 00    jmpq   *0x63fdc0(,%rax,8)
```
When slicing to 0x46cc94, dyninst will try to read the content of `push_default`, finds that the eax equals to a constant, which leads to wrong bounds of `rax`.

The binary of above example is attached:
[git_gcc_O2.gz](https://github.com/dyninst/dyninst/files/4073145/git_gcc_O2.gz)

**Comment by mxz297** [comment]
I am able to reproduce the bug. Will try to fix it. Thanks!

**Comment by mxz297** [comment]
Closed via #695
